### ISSUE TITLE
Nested routers w/ path parameters?

### ISSUE THREAD

**Issue by akotsampaseris** [vitalik/django-ninja#1055]
# Use Case

I have several objects that belong to a certain customer, who has a unique **customer_id**. I want to include **"/customers/{customer_id}/"** as a prefix for all subsequent routers that handle the objects that belong to that customer. And I need the **customer_id** to ensure that the user/agent requesting access to these objects belongs to the same customer.

# Example

**In expense/api.py:**

```python
router = Router(tags=["Expense"])
router.add_router("customers/{customer_id}/", products_router)
```

**In products/api.py:**
```python
router = Router(tags=["Products"])

@router.get("/products", response={200: List[Products], 400: Error})
def get_products(request: WSGIRequest, customer_id: int):
    products = ProductService.get_products()
    return 200, products
``` 

**Error Response:**
```javascript
{
    "detail": [
        {
            "loc": [
                "query",
                "customer_id"
            ],
            "msg": "field required",
            "type": "value_error.missing"
        }
    ]
}
``` 

# How I want to solve it

I have created a Router object for each Django app, eg app_router, and I am adding each of those routers to the main Router object, eg main_router, using **main_router.add_router("customers/{customer_id}", app_router)**. *_(The router names are used just as an example)._

# Problem

I need to access the **customer_id** as a path parameter in my views. The problem is that the view does not recognize it as a path parameter, because it belongs to a higher level component, even though the endpoint is created correctly. When I include **"/customers/{customer_id}/"** directly to the same exact view, it obviously sees the path parameter and it works correctly. But this is an issue when you have a big number of endpoints that you need to include that.

# Questions

I saw that FastAPI initially had a similar issue and the path parameters from nested routers were not included in the view level. But then they fixed that and included that functionality in a later release. Is there a way to do that with Django-Ninja already? If not, is it possible to include that to a later version? Is there a reason that this structure would not be advised? Thank you!

**Comment by eznix86** [comment]
Not sure, but Maybe try to create a [global auth](https://django-ninja.dev/guides/authentication/#global-authentication)

``` python
router = Router(tags=["Expense"], auth=AuthHere) # <---
router.add_router("customers/{customer_id}/", products_router)
```

Then inside of this
```diff
router = Router(tags=["Products"])

@router.get("/products", response={200: List[Products], 400: Error})
-def get_products(request: WSGIRequest, customer_id: int):
+def get_products(request: WSGIRequest):
+   user_id = request.auth
    products = ProductService.get_products()
    return 200, products
```

**Comment by akotsampaseris** [comment]
@eznix86 yeah, that works, but it defeats the purpose of the prefix, because I have to append it to every endpoint. But thanks for your help, bud!

**Comment by almeida-raphael** [comment]
I agree with @akotsampaseris  that this is a necessary feature for a future release

**Comment by vitalik** [comment]
Hi @akotsampaseris 

since router does not really know how it will be included you need to "help" it to understand where to get your customer_id parameter - try this:


```Python
from ninja import Path
router = Router(tags=["Products"])


def get_products(request: WSGIRequest, customer_id: int = Path(...)): # <--- Path
    ...
```

**Comment by vitalik** [comment]
or `customer_id: Path[int]` if you use ninja 1.x

**Comment by Marclev78** [comment]
What should go into the ... in `Path(...)`? I have run into the same thing and am trying to understand how to fix it.

**Comment by davitv** [comment]
@vitalik Maybe it’s worth mentioning the `Path`  type annotation on the nested router documentation page?
